### PR TITLE
HKG: Hyundai Ioniq 5 N 2024 Car Port

### PR DESCRIPTION
**Pull request by Kirito3481**
**Checklist**

- [x] added entry to CAR in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [ ] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [ ] route with openpilot:
- [x] route with stock system: `78ad5150de133637/2024-05-09--03-49-06`
- [x] car harness used (if comma doesn't sell it, put N/A): Hyundai M

**Issues**
- [ ] Harness Relay Malfunction occurs when OP Long is enabled.

### CODE REVIEW

**Comment by Kirito3481** [comment]
> Can this car be kept inside the `IONIQ_5` platform?

Is there a way to distinguish between regular ioniq5 and n in interface.py? Mass, wheelbase, and steerRatio are slightly different.
Or is it okay if it's slightly different?

**Comment by sshane** [comment]
The wheelbase looks identical (our 2.97 should be 2.99), mass differs due to the battery size (but it already doesn't match exactly for the various trims we support, so that's fine), so that all looks fine.

If we can't detect that turn signal difference yet, sounds like I didn't do any verification on it yet, then you can leave it as a separate platform. You should rebase before you perform any more changes.

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -538,11 +537,10 @@ def match_fw_to_car_fuzzy(live_fw_versions, offline_fw_versions) -> Set[str]:
 CANFD_CAR = {CAR.KIA_EV6, CAR.IONIQ_5, CAR.IONIQ_6, CAR.TUCSON_4TH_GEN, CAR.SANTA_CRUZ_1ST_GEN, CAR.KIA_SPORTAGE_5TH_GEN, CAR.GENESIS_GV70_1ST_GEN,
              CAR.GENESIS_GV60_EV_1ST_GEN, CAR.KIA_SORENTO_4TH_GEN, CAR.KIA_NIRO_HEV_2ND_GEN, CAR.KIA_NIRO_EV_2ND_GEN,
              CAR.GENESIS_GV80, CAR.KIA_CARNIVAL_4TH_GEN, CAR.KIA_SORENTO_HEV_4TH_GEN, CAR.KONA_EV_2ND_GEN, CAR.KIA_K8_HEV_1ST_GEN,
-             CAR.STARIA_4TH_GEN, CAR.IONIQ_5_N}
+             CAR.STARIA_4TH_GEN}
 
 # The radar does SCC on these cars when HDA I, rather than the camera
-CANFD_RADAR_SCC_CAR = {CAR.GENESIS_GV70_1ST_GEN, CAR.KIA_SORENTO_4TH_GEN, CAR.GENESIS_GV80, CAR.KIA_CARNIVAL_4TH_GEN, CAR.KIA_SORENTO_HEV_4TH_GEN,
-                       CAR.IONIQ_5_N}
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py]: I don't think this is true

**Comment by sshane** [comment]
You can add `HyundaiFlags.CANFD_NO_RADAR_DISABLE` for now

File: selfdrive/car/hyundai/carstate.py [hunk 2]
```diff
@@ -208,7 +208,7 @@ def update_canfd(self, cp, cp_cam):
 
     # TODO: alt signal usage may be described by cp.vl['BLINKERS']['USE_ALT_LAMP']
     left_blinker_sig, right_blinker_sig = "LEFT_LAMP", "RIGHT_LAMP"
-    if self.CP.carFingerprint == CAR.HYUNDAI_KONA_EV_2ND_GEN:
+    if bool(cp.vl['BLINKERS']['USE_ALT_LAMP']):
```
Comment (sshane) [inline selfdrive/car/hyundai/carstate.py:211]: can you make this a tuple check for the two cars? don't add a flag yet, as once we detect it, it won't require a flag

**Comment by thisLambo** [comment]
Just to add to the fingerprints, here's the fingerprint for the US `fwdCamera`
`b'\xf1\x00NE1NMFC  AT USA LHD 1.00 1.04 99211-NI000 231219'`

**Comment by Kirito3481** [comment]
> Just to add to the fingerprints, here's the fingerprint for the US `fwdCamera` `b'\xf1\x00NE1NMFC AT USA LHD 1.00 1.04 99211-NI000 231219'`

fwdRadar is same?

**Comment by thisLambo** [comment]
> > Just to add to the fingerprints, here's the fingerprint for the US `fwdCamera` `b'\xf1\x00NE1NMFC AT USA LHD 1.00 1.04 99211-NI000 231219'`
> 
> fwdRadar is same?

Yeah

> > Just to add to the fingerprints, here's the fingerprint for the US `fwdCamera` `b'\xf1\x00NE1NMFC AT USA LHD 1.00 1.04 99211-NI000 231219'`
> 
> fwdRadar is same?

Yeah, `b'\xf1\x00NE1N RDR -----      1.00 1.00 99110-NI000         '`

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**